clouddriver/clouddriver-artifacts/src/test/java/com/netflix/spinnaker/clouddriver/artifacts/docker/HelmOciDockerArtifactCredentialsTest.java

```diff
@@ -0,0 +1,331 @@
+            .build();
+
+    // Create credentials with mocked dependencies
+    credentials =
+        new HelmOciDockerArtifactCredentials(
+            account, new OkHttpClient(), helmOciFileSystem, serviceClientProvider);
+  }
+
+  private static <T> T buildService(Class<T> type, String baseUrl) {
+    return new Retrofit.Builder()
+        .baseUrl(baseUrl)
+        .client(new OkHttpClient())
+        .addCallAdapterFactory(ErrorHandlingExecutorCallAdapterFactory.getInstance())
+        .addConverterFactory(JacksonConverterFactory.create())
+        .build()
+        .create(type);
+  }
+
+  @Test
+  void testGetType() {
+    assertEquals("artifacts-helm-oci", credentials.getType());
+  }
+
+  @Test
+  void testGetArtifactNames() {
+    List<String> names = credentials.getArtifactNames();
+    assertEquals(List.of("test-repo", "another-repo"), names);
+  }
+
+  @Test
+  void testGetArtifactVersions() {
+    // Setup
+    String artifactName = "test-repo";
+    DockerRegistryTags tags = mock(DockerRegistryTags.class);
+    when(tags.getTags()).thenReturn(List.of("1.0.0", "1.1.0", "2.0.0"));
+
+    // Mock the WireMock server to return tags
+    wmDockerRegistry.stubFor(
+        get(urlPathMatching("/v2/" + artifactName + "/tags/list"))
+            .willReturn(
+                aResponse()
+                    .withStatus(200)
+                    .withHeader("Content-Type", "application/json")
+                    .withBody(
+                        "{\"name\":\""
+                            + artifactName
+                            + "\",\"tags\":[\"1.0.0\",\"1.1.0\",\"2.0.0\"]}")));
+
+    // Execute
+    List<String> versions = credentials.getArtifactVersions(artifactName);
+
+    // Verify
+    assertNotNull(versions);
+    assertEquals(3, versions.size());
+    assertTrue(versions.contains("1.0.0"));
+    assertTrue(versions.contains("1.1.0"));
+    assertTrue(versions.contains("2.0.0"));
+  }
+
+  @Test
+  void testDownloadArtifact() throws IOException, InterruptedException {
+    // Setup
+    String artifactName = "test-repo";
+    String version = "1.0.0";
+    Artifact artifact = Artifact.builder().name(artifactName).version(version).build();
+
+    // Create test content
+    String testContent = "test helm chart content";
+    File tempFile = tempDir.resolve("test-content.tgz").toFile();
+    FileUtils.writeStringToFile(tempFile, testContent, "UTF-8");
+
+    // Setup file system mocks
+    Path stagingPath = tempDir.resolve("test-repo-hash");
+    Path outputFile = stagingPath.resolve("1.0.0.tar.gz");
+    when(helmOciFileSystem.getLocalClonePath(artifactName, version)).thenReturn(stagingPath);
+    when(helmOciFileSystem.tryTimedLock(artifactName, version)).thenReturn(true);
+    when(helmOciFileSystem.canRetainClone()).thenReturn(false);
+
+    // Mock WireMock to return the getManifest response
+    wmDockerRegistry.stubFor(
+        get(urlPathMatching("/v2/" + artifactName + "/manifests/" + version))
```

Comment:
Called it out internall on a slack conversation... but making sure a sub folder based artifactName works is needed.  In case retrofit2 or something else might encode the "folders" for the artifactName in such a way to break this.